File: src/com/markupartist/sthlmtraveling/receivers/OnAlarmReceiver.java
package com.markupartist.sthlmtraveling.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.util.Log;

import com.markupartist.sthlmtraveling.service.DeviationService;
import com.markupartist.sthlmtraveling.service.WakefulIntentService;

public class OnAlarmReceiver extends BroadcastReceiver {
    private static final String TAG = "OnAlarmReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d(TAG, "recieved alarm");

        DeviationService.startService(context);
    }
  }

File: sthlmtraveling/src/main/java/com/markupartist/sthlmtraveling/receivers/OnAlarmReceiver.kt
package com.markupartist.sthlmtraveling.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.markupartist.sthlmtraveling.service.DeviationService.Companion.startService

class OnAlarmReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent?) {
        Log.d(TAG, "Received alarm")
        startService(context)
    }

    companion object {
        private const val TAG = "OnAlarmReceiver"
    }
}
